modules/file-transfer/src/main/java/org/apache/ignite/internal/network/file/FileTransferServiceImpl.java

```diff
@@ -0,0 +1,402 @@
+                    } else if (message instanceof FileHeaderMessage) {
+                        processFileHeader((FileHeaderMessage) message);
+                    } else if (message instanceof FileChunkMessage) {
+                        processFileChunk((FileChunkMessage) message);
+                    } else {
+                        LOG.error("Unexpected message received: {}", message);
+                    }
+                });
+    }
+
+    private void processFileTransferErrorMessage(FileTransferErrorMessage message) {
+        fileReceiver.receiveFileTransferErrorMessage(message);
+    }
+
+    private void processUploadRequest(FileUploadRequest message, String senderConsistentId, long correlationId) {
+        completedFuture(UUID.randomUUID())
+                .thenCompose(transferId -> {
+                    Path directory = createTransferDirectory(transferId);
+                    FileTransferMessagesHandler handler = fileReceiver.registerTransfer(
+                            senderConsistentId,
+                            transferId,
+                            directory
+                    );
+                    FileUploadResponse response = factory.fileUploadResponse()
+                            .transferId(transferId)
+                            .build();
+                    Metadata metadata = message.metadata();
+                    return messagingService.respond(senderConsistentId, FILE_TRANSFERRING_CHANNEL, response, correlationId)
+                            .thenCompose(ignored -> {
+                                return handler.result()
+                                        .thenCompose(files -> getFileHandler(metadata).handleUpload(metadata, files));
+                            })
+                            .whenComplete((v, e) -> {
+                                if (e != null) {
+                                    LOG.error(
+                                            "Failed to handle file upload. Transfer ID: {}. Metadata: {}. Exception: {}",
+                                            transferId,
+                                            metadata,
+                                            e
+                                    );
+                                }
+                                deleteDirectoryIfExists(directory);
+                            });
+                });
+    }
+
+    private void processDownloadRequest(FileDownloadRequest message, String senderConsistentId, Long correlationId) {
+        supplyAsync(() -> getFileProvider(message.metadata()))
+                .thenCompose(provider -> provider.files(message.metadata()))
+                .whenComplete((files, e) -> {
+                    if (e != null) {
+                        LOG.error("Failed to get files for download. Metadata: {}", message.metadata(), e);
+                        FileDownloadResponse response = factory.fileDownloadResponse()
```

Comment:
I will ask you one more time here to use more empty lines in your code. I didn't write this comment every time. Code this dense is difficult to read and it doesn't look like the rest of Ignite code. Please fix it everywhere in your code.